nuxeo-features/nuxeo-platform-imaging/nuxeo-platform-imaging-core/src/main/java/org/nuxeo/ecm/platform/picture/recompute/RecomputeViewsAction.java

```diff
@@ -0,0 +1,129 @@
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.EventService;
+import org.nuxeo.ecm.core.event.impl.DocumentEventContext;
+import org.nuxeo.ecm.platform.picture.api.adapters.PictureResourceAdapter;
+import org.nuxeo.lib.stream.computation.Topology;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.stream.StreamProcessorTopology;
+
+/**
+ * BAF Computation that fills picture views for the blob property described by the given xpath.
+ *
+ * @since 11.1
+ */
+public class RecomputeViewsAction implements StreamProcessorTopology {
+
+    public static final String ACTION_NAME = "recomputeViews";
+
+    public static final String PARAM_XPATH = "xpath";
+
+    @Override
+    public Topology getTopology(Map<String, String> options) {
+        return Topology.builder()
+                .addComputation(RecomputeViewsComputation::new, //
+                        Arrays.asList(INPUT_1 + ":" + ACTION_NAME, OUTPUT_1 + ":" + STATUS_STREAM))
+                .build();
+    }
+
+    public static class RecomputeViewsComputation extends AbstractBulkComputation {
+
+        public static final String PICTURE_VIEWS_GENERATION_DONE_EVENT = "pictureViewsGenerationDone";
+
+        protected String xpath;
+
+        public RecomputeViewsComputation() {
+            super(ACTION_NAME);
+        }
+
+        @Override
+        public void startBucket(String bucketKey) {
+            BulkCommand command = getCurrentCommand();
+            xpath = command.getParam(PARAM_XPATH);
+        }
+
+        @Override
+        protected void compute(CoreSession session, List<String> ids, Map<String, Serializable> properties) {
+
+            for (String docId : ids) {
+
+                if (!session.exists(new IdRef(docId))) {
+                    continue;
+                }
+
+                DocumentModel workingDocument = session.getDocument(new IdRef(docId));
+                Property fileProp = workingDocument.getProperty(xpath);
+                Blob blob = (Blob) fileProp.getValue();
+                if (blob == null) {
+                    // do nothing
+                    continue;
+                }
+
+                String title = workingDocument.getTitle();
+                try {
+                    PictureResourceAdapter picture = workingDocument.getAdapter(PictureResourceAdapter.class);
+                    picture.fillPictureViews(blob, blob.getFilename(), title, null);
+                } catch (DocumentNotFoundException e) {
+                    // a parent of the document may have been deleted.
+                    continue;
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+
+                if (workingDocument.isVersion()) {
+                    workingDocument.putContextData(ALLOW_VERSION_WRITE, Boolean.TRUE);
+                }
+                workingDocument.putContextData("disableNotificationService", Boolean.TRUE);
```

Comment:
There is the constant `NotificationConstants.DISABLE_NOTIFICATION_SERVICE` that can be used.

Reply:
Not in the classpath